addons/purchase_stock/models/account_move_line.py

```diff
@@ -77,72 +77,10 @@ def _generate_price_difference_vals(self, layers):
         # That way, we ensure a deterministic behaviour
         history.sort(key=lambda item: (item[0], bool(item[1]), (item[1] or item[2]).id))
 
-        # the next dict is a matrix [layer L, invoice I] where each cell gives two info:
-        # [initial qty of L invoiced by I, remaining invoiced qty]
-        # the second info is usefull in case of a refund
-        layers_and_invoices_qties = defaultdict(lambda: [0, 0])
-
-        # the next dict will also provide two info:
-        # [total qty to invoice, remaining qty to invoice]
-        # we need the total qty to invoice, so we will be able to deduce the invoiced qty before `self`
-        qty_to_invoice_per_layer = defaultdict(lambda: [0, 0])
-
         # Replay the whole history: we want to know what are the links between each layer and each invoice,
         # and then the links between `self` and the layers
         history.append((False, self, False))  # time was only usefull for the sorting
```

Comment:
Why don't you move that into `_replay_history` as well, as it's only useful there?

Reply:
For me it made sense to keep the whole "history building" together. I guess this kind of sits between that and replaying the history so yeah